scripts/validate-schemas.js

```diff
@@ -1,90 +1,71 @@
-                validate = ajvDraft04.validateSchema(obj);
-                if(validate){
-                    console.log(`\n${file}: JSON Schema is valid!`);
-                }   
-            } else {
-                // Validate the schema
-                validate = ajv.validateSchema(obj);
-                if(validate){
-                    console.log(`\n${file}: JSON Schema is valid!`);
-                }
-            }
-    
-            // Check if the schema is not valid and collect errors
-            if (!validate) {
-                validationErrors.push({
-                    file,
-                    errors: obj.$schema === 'http://json-schema.org/draft-04/schema'
-                    ? ajvDraft04.errors
-                    : ajv.errors
-                });
-            }
-        } catch (error) {
-            validationErrors.push({
-            file,
-            errors: [{ message: `\nError reading or parsing JSON Schema: ${error.message}` }]
-            });
-        }
-        });
-
-            // Print errors after processing all files
-        validationErrors.forEach(({ file, errors }) => {
-            console.error(`\n${file}: JSON Schema is not valid:`, errors);
-        });
-
-        // Exit with an error code if there are validation errors
-        if (validationErrors.length > 0) {
-            process.exit(1);
+    filteredFiles.forEach(file => {
+      const filePath = path.join(directoryPath, file);
+
+      const schemaValidator = (obj) => {
+        return (obj.$schema === 'http://json-schema.org/draft-04/schema') ? ajvDraft04.validateSchema(obj) : ajv.validateSchema(obj);
+      };
+
+      const validationResult = validateSchema(filePath, fs.readFileSync(filePath, 'utf8'), schemaValidator);
+
+      if (!validationResult.validate || validationResult.errors.length > 0) {
+        validationErrors.push(validationResult);
+      }
+    });
+
+    if (validationErrors.length > 0) {
+      console.error('\nValidation errors:');
+      validationErrors.forEach(({ filePath, validate, errors }) => {
+        console.error(`${filePath}: JSON Schema is not valid:`);
+
+        if (validate) {
+          console.error('Detailed Error Information:');
+          errors.forEach(error => {
+            console.error(JSON.stringify(error, null, 2));
+          });
+        } else {
+          console.error(errors);
         }
-        
-    } catch (error) {
-        console.error('\nError during validation:', error.message);
-        process.exit(1);
+      });
+      process.exit(1);
     }
+  } catch (error) {
+    console.error('\nError during validation:', error.message);
+    process.exit(1);
+  }
 }
 
-
-const excludedFiles=['2.0.0-rc1.json', '2.0.0-rc1-without-$id.json']; // added temporarily to avoid validation failure due to these two files. The schemas version are incorrect in these and needs to be fixed.
-
+const excludedFiles = ['2.0.0-rc1.json', '2.0.0-rc1-without-$id.json'];
```

Comment:
please keep the comment for future reference to other people who may see/modify the code

Reply:
Done @AnimeshKumar923